### ISSUE TITLE
Rule proposal: `regex-strict-mode`

### ISSUE THREAD

**Issue by Nantris** [sindresorhus/eslint-plugin-unicorn#1500]
This rule would be beneficial to allow users to get the benefits of `unicorn/better-regex` without interfering with the [`regexp/strict` rule](https://ota-meshi.github.io/eslint-plugin-regexp/rules/strict.html) from the [`eslint-plugin-regexp`](https://github.com/ota-meshi/eslint-plugin-regexp) project.

## Fail

```js
const demo = /\}\{/ // Flagged by unicorn/better-regex if it has the unnecessary escape characters - and by `regexp/strict` if it doesn't.
```

## Pass

Pass would be leaving the code untouched and without warning, as strict mode encourages including the escape characters. This also has the benefit of improving syntax highlighting, as the `{` and `}` are clearly highlighted as characters to match, as opposed to being highlighted as quantifiers.

```js
const demo = /\}\{/ // Same code, just don't warn
```

**Comment by papb** [comment]
Your request is actually for an option to `better-regex`, not for a new rule, right?

**Comment by Slapbox** [comment]
I suppose that's accurate @papb.

**Comment by sindresorhus** [comment]
It was removed: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/89dc5dc04389929ae786e171bac2af522fe2e7d8
